representation/knob-representation.metta

```diff
@@ -60,8 +60,41 @@
 (= (getKnobSpec (mkLSK (mkDiscKnob $knob (mkMultip $multip) $curr $def $dis) $subtree))
 (mkDiscSpec (- $multip (List.length $dis))))
 
+;; Extracts the location from a discrete or a logical subtree knob.
+;; Params:
+;;  $knob: Either discrete or logical subtree knob
+;; Returns: The location of the knob.
+(: getKnobLoc (-> DiscreteKnob (NodeId)))
+(: getKnobLoc (-> LogicalSubtreeKnob (NodeId)))
+(= (getKnobLoc (mkDiscKnob (mkKnob $tree $loc) $multip $curr $def $dis)) $loc)
+(= (getKnobLoc (mkLSK $discKnob $subtree)) (getKnobLoc $discKnob))
+
 ;; A comparison function for discSpec 
 ;; (used as a $compFunc to insert knobSpec and knob pairs into a multimap ) 
 (: discSpec<= (-> DiscSpec DiscSpec Bool))
 (= (discSpec<= (mkDiscSpec $newKey) (mkDiscSpec $currKey))
 (<= $newKey $currKey))
+
+;; Checks of two nodeIds are equal.
+;; Params: $id1 & $id2
+;; Returns: True/False
+(: nodeId== (-> NodeId NodeId Bool))
+(= (nodeId== (mkNodeId $id1) (mkNodeId $id2))
```

Comment:
What is the difference between this code and using `==`? I think they have the same effect right?